src/foam/core/Slot.js

```diff
@@ -642,7 +642,7 @@ foam.CLASS({
       if ( arguments.length != 1 ) return this.SUPER.apply(this, arguments);
       var subs = this.slots.map(s => s.sub(l));
```

Comment:
Could we instead remove the edit on line 645 and instead change this line to:
var subs = this.slots.map(s => s && s.sub(l));

Then we don't waste space on null values?